homeassistant/components/proxmoxve/__init__.py

```diff
@@ -145,41 +103,100 @@ def build_client() -> ProxmoxAPI:
         host_name = host_config["host"]
         coordinators[host_name] = {}
 
-        proxmox_client = hass.data[PROXMOX_CLIENTS][host_name]
+    # import to config flow
+    if DOMAIN in config:
+        for conf in config[DOMAIN]:
+            _LOGGER.warning(
+                # Proxmox VE config flow added in 2022.8 and should be removed in 2022.10
+                "Configuration of the Proxmox in YAML is deprecated and "
+                "will be removed in Home Assistant 2022.10; Your existing configuration "
+                "has been imported into the UI automatically and can be safely removed "
+                "from your configuration.yaml file"
+            )
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+                )
+            )
 
-        # Skip invalid hosts
-        if proxmox_client is None:
-            continue
+    return True
 
-        proxmox = proxmox_client.get_api_client()
 
-        for node_config in host_config["nodes"]:
-            node_name = node_config["node"]
-            node_coordinators = coordinators[host_name][node_name] = {}
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the platform."""
 
-            for vm_id in node_config["vms"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, vm_id, TYPE_VM
-                )
+    entry_data = config_entry.data
 
-                # Fetch initial data
-                await coordinator.async_refresh()
+    hass.data[PROXMOX_CLIENTS] = {}
```

Comment:
We shouldn't be storing integration-level objects at this high of a level within `hass.data`. The common pattern is `hass.data[DOMAIN][CONFIG_ENTRY_ID] = {}`.

Reply:
Removed duplicates between async_setup and async_setup_entry, removing this storage.